### DISCUSSION TITLE
meshing in porous flow module

### DISCUSSION THREAD

**Discussion by lupeom** [idaholab/moose#16897] [Q&A Meshing] [answered]
Hello moose community,
I am new comer to moose and i would like to use porous flow to simulation the injection of CO2 in brine-saturated reservoir. I found all the geometry/meshes in the tutorials are horizontal and regular geometry. I am kindly wondering if it possible to include some titling meshed geometry-- a case where the mesh has a fold- (or dome-) shape? Can we do that in moose or one needs a third-party software?
thanks and I will much appreciate your help in advance.
Best, Lupeom

**Comment by WilkAndy** [comment, marked as answer]
Hi Lupeom,

Almost all of the PorousFlow tests and examples use MOOSE's inbuilt meshing capability.  While that is suitable for highly idealised situations, you almost always need something more sophisticated when dealing with real geology and hydrostratigraphy, as you have found.

You will need to use an external meshing software package.  I have built many models of the subsurface, and have found that meshing often takes a significant fraction of the time spent modelling, so don't think that meshing is a trivial, quick task!

Some people can get [Cubit](https://cubit.sandia.gov/) free, while others (like me) have to pay lots of $$$ for it.  There is a "sibling" product called [coreform](https://coreform.com/products/coreform-cubit/) that has free aspects.  Many MOOSE people use the excellent and free tool called [gmsh](https://gmsh.info/) that works well for PorousFlow type of problems.  If you are going to be in this field for many years, you might like to make your own mesh-translation tools (between different formatting) and mesh-building tools, probably based on the exodus or VTK libraries, since none of the aforementioned packages easily handle subsurface hydrostratigraphy with faults and so on.

a

**Reply by aeslaughter** [reply]
Coreform Cubit is the commercial version of Cubit: https://cubit.sandia.gov/licensing.html

**Comment by GregVernon** [comment]
@lupeom

As mentioned above, we (Coreform) provide a non-commercial license for Coreform Cubit.  It's fully-functional but is limited to exporting 50k elements.  If this limit is too restricting, contact us and we can discuss temporarily removing this restriction (especially if it's for a dissertation or thesis).  If you're interested in the Coreform Cubit route, you can find [Cubit tutorials on our website](https://coreform.com/products/coreform-cubit/tutorials/) and you can post Cubit-specific questions on our [forum](forum.coreform.com) for the Cubit community to answer.

**Reply by liaohaiyang1534** [reply]
Can I apply for the temporary removal of the export restriction? I am writing a research paper on numerical simulations.
Thanks!

**Comment by lupeom** [comment]
Thanks so much @WilkAndy and @GregVernon  for your valuable comments, so much appreciated.

**Reply by mcacace** [reply]
@lupeom you could also try our open source meshing generator (look here https://github.com/idaholab/moose/discussions/16537#discussion-75291). It has been developed for similar applications as those you are interested and it is fully Moose compatibile. Let me know if you would have any questions

**Reply by lupeom** [reply]
Thanks so much @mcacace, very interesting.
